### PR TITLE
Add code and json for problem_check_fail event

### PR DESCRIPTION
**Pull request by danialmalik**
**Trello Link:** _https://trello.com/c/kebEgzMI/20-problem-interaction-event-problemcheckfail_

**Description:** 
The server emits problem_check_fail events when a problem cannot be
checked successfully.
Refactored some code for `problem_check` event.

**Checks before merge:**

- [ ] Reviewed
- [ ] Commits squashed

### CODE REVIEW

File: openedx/features/caliper_tracking/transformers/problem_transformers.py [hunk 1]
```diff
@@ -194,60 +193,33 @@ def problem_check(current_event, caliper_event):
...
-    caliper_event['extensions']['extra_fields'].update({
-        'course_id': current_event['context']['course_id'],
-        'ip': current_event['ip']
-    })
-    return caliper_event
-
+    caliper_event['extensions']['extra_fields'].update(current_event['context'])
+    caliper_event['extensions']['extra_fields']['ip'] = current_event['ip']
 
-def problem_check_server(current_event, caliper_event):
```
Comment (asamolion) [inline openedx/features/caliper_tracking/transformers/problem_transformers.py]: Why these changes?

Comment (danialmalik) [reply openedx/features/caliper_tracking/transformers/problem_transformers.py]: Because we can use a single method for three events. Previously there were two methods to a single difference in two event logs. Now both can be transformed using this transformer and also this one.

Comment (asamolion) [reply openedx/features/caliper_tracking/transformers/problem_transformers.py]: Alright.
